central_stats/main.bal

```diff
@@ -0,0 +1,177 @@
+let ballerinaCount = mainTable_ballerina
+| summarize org = "ballerina" ,Count = tostring(count()) ;
+let ballerinaxCount = mainTable_ballerinax
+| summarize org = "ballerinax" ,Count = tostring(count()) ;
+let result = union ballerinaCount , ballerinaxCount ;
+result;`;
+
+string queryPullCountByCountry = string `let mainTable = customEvents
+| where timestamp > ago(${timeDuration})
+| where name == "package-pull";
+let x = mainTable
+| summarize Count = tostring(count()) by client_CountryOrRegion, Event = "package-pull" ;
+x ;`;
+
+string queryPushCount = string `let mainTable = customEvents
+| where timestamp > ago(${timeDuration})
+| where name == "package-push" ;
+let x = mainTable
+| summarize Count = tostring(count()) by client_CountryOrRegion, Event = "package-push" , package = tostring(customDimensions["name"]) , org = tostring(customDimensions["organization"]) ;
+x ;`;
+
+string queryDistDownloadCount = string `let mainTable = customEvents
+| where timestamp > ago(${timeDuration})
+| where name == "distribution-download";
+let y = mainTable
+| summarize Count = tostring(count()) by client_CountryOrRegion, Event = "distribution-download" , Version = tostring(customDimensions["downloadedDistVersion"]) ;
+y;`;
+
+string queryPackages = string `let mainTable = customEvents
+| where timestamp > ago(${timeDuration})
+| where name == "package-pull";
+let x = mainTable
+| summarize PullCount = tostring(count()) by tostring(customDimensions["name"]) ,  org = tostring(customDimensions["organization"])
+| sort by PullCount desc;
+x ;`;
+
+configurable string SPREADSHEET_CLIENT_ID = os:getEnv("SPREADSHEET_CLIENT_ID");
+configurable string SPREADSHEET_CLIENT_SECRET = os:getEnv("SPREADSHEET_CLIENT_SECRET");
+configurable string SPREADSHEET_REFRESH_TOKEN = os:getEnv("SPREADSHEET_REFRESH_TOKEN");
+configurable string SPREADSHEET_ID = os:getEnv("SPREADSHEET_ID");
+configurable string applicationID = os:getEnv("APPLICATION_ID");
+configurable string apiKey = os:getEnv("API_KEY");
+
+public type HttpResponse record {
+    Tables[] tables;
+};
+
+public type Tables record {
+    string name;
+    Column[] columns;
+    string[][] rows;
+};
+
+public type Column record {
+    string name;
+    string 'type;
+};
+
+public function main() returns error? {
+
+    http:Client http = check new http:Client(string `https://api.applicationinsights.io/v1/apps/${applicationID}`);
+
+    sheets:ConnectionConfig spreadsheetConfig = {
+        auth: {
+            clientId: SPREADSHEET_CLIENT_ID,
+            clientSecret: SPREADSHEET_CLIENT_SECRET,
+            refreshUrl: sheets:REFRESH_URL,
+            refreshToken: SPREADSHEET_REFRESH_TOKEN
+        }
+    };
+
+    sheets:Client spreadsheetClient = check new (spreadsheetConfig);
+
+    // pull count - ballerina/ballerinax
+    string encodedQuery = check url:encode(queryPullCountOfBallerinaBallerinax, "UTF-8");
+    HttpResponse response = check http->/query.get({
+            "x-api-key": apiKey
```

Comment:
Move `x-api-key` to a constant.

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-release/pull/2576/commits/b351f15e1d0a2d11b1ce8f7b32e100830175825d